include/mlir/Dialect/StandardOps/Ops.td

```diff
@@ -820,6 +820,45 @@ def SelectOp : Std_Op<"select", [NoSideEffect, SameOperandsAndResultShape]> {
 
   let hasFolder = 1;
 }
+
+def SignExtendIOp : Std_Op<"sexti", [NoSideEffect]> {
```

Comment:
I think you're need SameOperandsAndResultShape here?

(assuming you support more than scalar)

Reply:
Agreed.